R/codemeta_description.R

```diff
@@ -48,134 +56,206 @@ additional_codemeta_terms <- function(){
-      }else{
-        # try to identify a GitHub or Gitlab repo
-        actual_code_repo <- code_repo[grepl("github\\.com", code_repo)|
-                                        grepl("gitlab\\.com", code_repo)][1]
-        # no direct link to README please
-        actual_code_repo <- gsub("#.*", "", actual_code_repo)
-        # otherwise take the first URL arbitrarily
-        if(is.null(codemeta$Repository)){
-          codemeta$codeRepository <- actual_code_repo
-        }
-
-        # add other URLs as related links
-        codemeta$relatedLink <- unique(c(codemeta$relatedLink,
-                                         code_repo[code_repo != actual_code_repo]))
-      }
-    }
+  # Store the package name in its own variable as it is used more than once
+  package_name <- descr$get("Package")
 
-    issue_tracker <- descr$get("BugReports")
-    if (!is.na(issue_tracker)){
-      codemeta$issueTracker <- issue_tracker
-    }
+  ## FIXME define an S3 class based on the codemeta list of lists?
+
+  if (is.null(id)) {
+    id <- package_name
+  }
 
+  if (is_IRI(id)) {
+    codemeta$`@id` <- id
+  }
 
+  codemeta$identifier <- package_name
+  codemeta$description <- descr$get("Description")
+  codemeta$name <- paste0(package_name, ": ", descr$get("Title"))
 
-    ## According to crosswalk, codemeta$dateModified and
-    ## codemeta$dateCreated are not crosswalked in DESCRIPTION
-    codemeta$datePublished <- NULL
-
-    codemeta$license <- spdx_license(descr$get("License"))
-
-    codemeta$version <- as.character(descr$get_version())
-    codemeta$programmingLanguage <-
-      list(
-        "@type" = "ComputerLanguage",
-        name = R.version$language,
-        version = paste(R.version$major, R.version$minor, sep = "."),
-        # According to Crosswalk, we just want numvers and not R.version.string
-        url = "https://r-project.org"
-      )
-    ## According to schema.org, programmingLanguage doesn't have a version;
-    ## but runtimePlatform, a plain string, does.
-    codemeta$runtimePlatform <- R.version.string
-
-    if (is.null(codemeta$provider))
-      codemeta$provider <- guess_provider(descr$get("Package"))
-    author <- try(descr$get_authors(), silent = TRUE)
-    if (!inherits(author,'try-error')) {
-      codemeta <-
-        parse_people(author, codemeta)
-    } else {
-      # get author and maintainer from their fields
-      # and don't get maintainer twice!
-      author <- as.person(descr$get("Author"))
-      maintainer <- descr$get_maintainer()
-      maintainer <- as.person(paste(maintainer))
-      maintainer$role <- "cre"
-      author_strings <- paste(author$given, author$family)
-      maintainer_strings <- paste(maintainer$given, maintainer$family)
-      author <- author[!author_strings %in% maintainer_strings]
-
-      author <- c(author, maintainer)
-      codemeta <-
-        parse_people(author, codemeta)
-    }
+  ## add repository related terms
+  codemeta <- add_repository_terms(codemeta, descr)
```

Comment:
Since you created helper functions, could you add tests for them in the file with codemeta_description tests?

Reply:
Yes, I will add some simple tests.